MapboxNavigation/NavigationViewController.swift

```diff
@@ -389,191 +379,138 @@ open class NavigationViewController: UIViewController {
         
-        let mapViewController = RouteMapViewController(routeController: self.routeController, delegate: self)
-        self.mapViewController = mapViewController
-        mapViewController.destination = route.legs.last?.destination
-        mapViewController.willMove(toParent: self)
-        addChild(mapViewController)
-        mapViewController.didMove(toParent: self)
-        let mapSubview: UIView = mapViewController.view
+        self.directions = directions
+        self.voiceController = voiceController
+        self.mapViewController = RouteMapViewController(routeController: self.routeController)
+        self.locationManager = NavigationLocationManager()
+		
+        super.init(nibName: nil, bundle: nil)
+		
+        self.mapViewController.delegate = self
+        self.mapViewController.willMove(toParent: self)
+        self.addChild(self.mapViewController)
+        self.mapViewController.didMove(toParent: self)
+        let mapSubview: UIView = self.mapViewController.view
         mapSubview.translatesAutoresizingMaskIntoConstraints = false
-        view.addSubview(mapSubview)
-        
+        self.view.addSubview(mapSubview)
         mapSubview.pinInSuperview()
-        mapViewController.reportButton.isHidden = !self.showsReportFeedback
         
         self.styleManager = StyleManager(self)
         self.styleManager.styles = [dayStyle, nightStyle]
-
-        if !(route.routeOptions is NavigationRouteOptions) {
-            print("`Route` was created using `RouteOptions` and not `NavigationRouteOptions`. Although not required, this may lead to a suboptimal navigation experience. Without `NavigationRouteOptions`, it is not guaranteed you will get congestion along the route line, better ETAs and ETA label color dependent on congestion.")
-        }
+        
+        self.mapViewController.navigationView.hideUI(animated: false)
+        self.mapView.tracksUserCourse = false
     }
     
+    convenience init() {
+        self.init(dayStyle: Style(demoStyle: ()))
+    }
+	
     deinit {
-        suspendNotifications()
+        self.suspendNotifications()
     }
     
+    // MARK: - UIViewController
+	
+    override open var preferredStatusBarStyle: UIStatusBarStyle {
+        self.currentStatusBarStyle
+    }
+	
     override open func viewDidLoad() {
         super.viewDidLoad()
         // Initialize voice controller if it hasn't been overridden.
         // This is optional and lazy so it can be mutated by the developer after init.
         _ = self.voiceController
         self.resumeNotifications()
-        view.clipsToBounds = true
+        self.view.clipsToBounds = true
     }
     
-    override open func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
+    // MARK: - NavigationViewController
+	
+    public func startNavigation(with route: Route, routeController: RouteController? = nil, locationManager: NavigationLocationManager = NavigationLocationManager()) {
```

Comment:
This routeController parameter is not currently being used. Did you intend to assign it?

And if so, does it even make sense to have a routeController parameter for init?